CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+# -----------------------------------------------------------------------------
+# Codam Coding College, Amsterdam @ 2022 by W2Wizard.
+# See README in the root project for more information.
+# -----------------------------------------------------------------------------
+
+# CMake specifications
+# -----------------------------------------------------------------------------
+cmake_minimum_required (VERSION 3.23.0 FATAL_ERROR)
+project(mlx42 VERSION 2.3.0)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Variables
+# -----------------------------------------------------------------------------
+set(SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(TOOLS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/tools)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+set(CMAKE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# Compile Options
+# -----------------------------------------------------------------------------
+# Reduce the size of LodePNG, we don't need these things.
+add_definitions(-D LODEPNG_NO_COMPILE_ENCODER)
+add_definitions(-D LODEPNG_NO_COMPILE_ANCILLARY_CHUNKS)
+
+if(UNIX)
+    set(CCSHADER ${PROJECT_SOURCE_DIR}/tools/compile_shader.sh)
+    add_compile_options(
+        -Wextra
+        -Wall
+        # -Werror # Uncomment this when im sure which warnings are important.
+        -Wunreachable-code
+		# Some low priority warnings that are annoying.
+        -Wno-char-subscripts
+        -Wno-sign-compare
+		-Wno-unused-parameter
+		-Wno-missing-field-initializers
+    )
+else()
+    set(CCSHADER ${PROJECT_SOURCE_DIR}/tools/compile_shader.bat)
+    add_compile_options(
+        /W4 # TODO: Figure out the different flags for windows.
+        /WX
+    )
+endif()
+
+# Build specific files
+# @see https://cmake.org/cmake/help/latest/command/add_custom_command.html
+# -----------------------------------------------------------------------------
+
+message(STATUS "MLX42 @ ${CMAKE_PROJECT_VERSION}")
+
+add_custom_command(
+    COMMENT "Building fragment shader"
+    DEPENDS ${CMAKE_SOURCE_DIR}/shaders/default.frag
+    OUTPUT mlx_frag_shader.c
+    COMMAND ${CCSHADER} ${CMAKE_SOURCE_DIR}/shaders/default.frag > mlx_frag_shader.c
```

Comment:
Instead of using `${CMAKE_SOURCE_DIR}` you should use `${CMAKE_CURRENT_SOURCE_DIR}` or `${PROJECT_SOURCE_DIR}` so one can compile it while having for example another CMakeLists.txt file in a parent directory